framework/test-vectors/README.md

```diff
@@ -0,0 +1,113 @@
+[//]: # "Copyright Amazon.com Inc. or its affiliates. All Rights Reserved."
+[//]: # "SPDX-License-Identifier: CC-BY-SA-4.0"
+
+## Summary
+
+These test vectors are intended to be used to validate interoperability
+across implementations of clients in a runtime,
+for the AWS Crypto Tools AWS Cryptographic Material Providers Library (MPL).
+
+They are composed of JSON manifest files that define one or more test cases,
+including sufficient information to process each test case.
+These manifest files can also identify additional resources needed for a given test case.
+These resources will be identified with a URI.
+If identifying a local file, the URI will be a relative path from the manifest file's
+parent directory to the target file.
+
+Every manifest type definition must include a unique type name that will be used by
+test vector handlers to identify the manifest type.
+
+See https://github.com/awslabs/aws-crypto-tools-test-vector-framework for additional information.
+This framework is much of the motivation for this work.
+
+The manifest is meant to produce encryption and decryption materials that every
+runtime implementation can process and produce the desired output for that test vector.
+
+These test vectors are not meant to be processed by a top level client library like
+the AWS Encryption SDK or the AWS Database Encryption SDK. However; they may be able
+to use the manifest to see how to construct keyrings that they can use in their test
+vectors.
+
+```mermaid
```

Comment:
it might be helpful to clarify/distinguish between test vector generation and test vector execution in these flow charts